### PR TITLE
Update Kubernetes support to Kubernetes v1.24.0

### PR DESCRIPTION
**Pull request by viveklak**

### CODE REVIEW

**Comment by viveklak** [comment]
@lblackstone looks like v1alpha1:CSIStorageCapacity and v1alpha1:CSIStorageCapacityList are no longer being served in 1.24 - https://github.com/kubernetes/kubernetes/pull/109151#discussion_r839535171. Do we need to add it to the deprecated.go file?

File: Makefile [hunk 1]
```diff
@@ -32,7 +32,7 @@ ensure::
 	cd tests && go mod tidy
 
 k8sgen::
-	(cd provider && CGO_ENABLED=0 go build -o $(WORKING_DIR)/bin/${CODEGEN} -ldflags "-X ${PROJECT}/${VERSION_PATH}=${VERSION}" ${PROJECT}/${PROVIDER_PATH}/cmd/$(CODEGEN))
+	(cd provider && CGO_ENABLED=1 go build -o $(WORKING_DIR)/bin/${CODEGEN} -ldflags "-X ${PROJECT}/${VERSION_PATH}=${VERSION}" ${PROJECT}/${PROVIDER_PATH}/cmd/$(CODEGEN))
```
Comment (viveklak) [inline Makefile:35]: Looks like the github.com/rjeczalik/notify CGO switch is also affecting us here.

Comment (stack72) [reply Makefile:35]: We will need to update goreleaser here

Comment (viveklak) [reply Makefile:35]: Maybe we pull this out and make the fixes to the build system separately rather than getting stuck behind the 1.24.0 release?

**Comment by lblackstone** [comment]
> @lblackstone looks like v1alpha1:CSIStorageCapacity and v1alpha1:CSIStorageCapacityList are no longer being served in 1.24 - [kubernetes/kubernetes#109151 (comment)](https://github.com/kubernetes/kubernetes/pull/109151#discussion_r839535171). Do we need to add it to the deprecated.go file?

Yeah, we should update the deprecation logic, and also want to make sure that thew v1alpha1 version is included in the list of aliases. It looks like it's not because the API was removed, so we'll need to patch that in.

**Comment by viveklak** [comment]
> > @lblackstone looks like v1alpha1:CSIStorageCapacity and v1alpha1:CSIStorageCapacityList are no longer being served in 1.24 - [kubernetes/kubernetes#109151 (comment)](https://github.com/kubernetes/kubernetes/pull/109151#discussion_r839535171). Do we need to add it to the deprecated.go file?
> 
> Yeah, we should update the deprecation logic, and also want to make sure that thew v1alpha1 version is included in the list of aliases. It looks like it's not because the API was removed, so we'll need to patch that in.

Yeah we should add the alias. Sounds good, will do! We still have time it seems for the 1.24.0 release (May 3rd)

**Review by lblackstone — APPROVED** [review]

**Pull request merged**